apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx

```diff
@@ -81,14 +84,46 @@ class LibraryCreationDialog extends React.Component {
       this.state.libraryClientApi,
       error =>
         this.setState({dialogState: DialogState.ERROR, errorMessage: error}),
-      libraryDetails =>
-        this.setState({
-          dialogState: DialogState.DONE_LOADING,
-          libraryDetails: libraryDetails
-        })
+      this.onLibraryLoaded
     );
   };
 
+  onLibraryLoaded = async libraryDetails => {
+    const defaultNewState = {
+      dialogState: DialogState.DONE_LOADING,
+      libraryDetails
+    };
+
+    try {
+      const profaneWords = await this.findProfanity(
+        libraryDetails.librarySource
+      );
+      if (profaneWords) {
```

Comment:
I'm a little worried about this case:

![image](https://user-images.githubusercontent.com/1615761/79479472-a9a80d00-7fc1-11ea-857c-9b5251d55628.png)

Especially since it's unclear what `findProfanity` may return just reading the code in this file - it's dependent on the server's behavior.  I'm worried that a future us might update that API to return an empty array when there's no profanity - which seems reasonable - and accidentally break this.